### PR TITLE
Branch02 building conn between FE and BE

### PR DESCRIPTION
**Pull request by MilosJovanovic1986**
**Motivation:** I'm trying to build a connection between `backend` and `frontend`. To build a connection between I'll use [socket.io](https://socket.io/)

### CODE REVIEW

File: backend/drone.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+const HOST = "192.168.10.1";
+const drone = dgram.createSocket("udp4");
+drone.bind(PORT);
+
+function parseState(state) {
+  return state.split(":").map((x) => x.split(":"));
+}
+
+const droneState = dgram.createSocket("udp4");
+droneState.bind(8890);
```
Comment (masone) [inline backend/drone.js:19]: this port is different from the `PORT`above. do you need both or can one be cleaned up maybe?

File: backend/drone.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+drone.on("message", (message) => {
+  console.log(`--> : ${message}`);
+
+  // io.sockets.emit('status', message.toString());
```
Comment (masone) [inline backend/drone.js:24]: do you still need this? if not, I'd delete it in the final version :)

File: backend/drone.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+function handleError(err) {
+  if (err) {
+    console.log("ERROR");
+    console.log(err);
+  }
+}
+
+const commands = ["command", "battery?", "takeoff", "land", "flip l", "down x"];
+
+const i = 0;
```
Comment (masone) [inline backend/drone.js:36]: if you're not actively relying on this variable, can you remove it?

File: backend/drone.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+drone.send("command", 0, "command".length, PORT, HOST, handleError);
```
Comment (masone) [inline backend/drone.js:38]: what's the purpose of `"command".length`?

Comment (masone) [reply backend/drone.js:38]: I believe this whole line might have been for debugging purposes, but has no actual value? 🤔

File: backend/package.json [hunk 5]
```diff
@@ -4,9 +4,17 @@
...
-  "license": "ISC"
+  "license": "ISC",
+  "dependencies": {
+    "express": "^4.17.1",
+    "lodash": "^4.17.21",
+    "socket.io": "^2.2.0",
+    "waait": "^1.0.5"
+  },
+  "devDependencies": {
+    "nodemon": "^1.18.7"
```
Comment (masone) [inline backend/package.json:18]: it does not look like you're using this (yet?). consider removing it

File: frontends/.next/build-manifest.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (masone) [inline frontends/.next/build-manifest.json:1]: the `.next` folder is a build folder from nextjs. please research how they recommend handling it when it comes to checking it in to git

File: frontends/components/DroneState.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import socket from "../socket";
+import { useState, useEffect } from "react";
+
+function useDroneState() {
+  const [droneState, updateDroneState] = useState([]);
+  useEffect(() => {
+    console.log("hey");
```
Comment (masone) [inline frontends/components/DroneState.js:7]: a lot of console logs :)
please clean them out before the final version

File: frontends/components/DroneState.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+}
+//     socket.on('status', message => {
+//         console.log('Message from socket');
+//         updateStatus(message);
+//     });
+//     return status;
+// }
+const DroneState = () => {
+  const status = useSocket();
+  const droneState = useDroneState([]);
```
Comment (masone) [inline frontends/components/DroneState.js:29]: the `useDroneState` does not take any attributes, but on this line, we're passing an empty array

**Review by masone — COMMENTED** [review]
Nice to get a first glimpse at your work from the last days! It's clearly still early for a proper review, so most of the comments might have become obvious to you sooner or later anyway.

Please check how to deal with the `.next` directory. it generates a lot of noise and makes it hard to review

File: backend/drone.js (hunk 1 above)
Comment (MilosJovanovic1986) [reply backend/drone.js:19]: I need this port to send information for `droneState`
state port

File: backend/drone.js (hunk 2 above)
Comment (MilosJovanovic1986) [reply backend/drone.js:24]: on the last branch already removed. Sorry for the mess on the lass branch..

File: backend/drone.js (hunk 3 above)
Comment (MilosJovanovic1986) [reply backend/drone.js:36]: the same here...

File: backend/drone.js (hunk 4 above)
Comment (MilosJovanovic1986) [reply backend/drone.js:38]: the purpose of the `command.length` is that we need to tell the drone how many characters the command content, when I remove the `command.length` then the drone will not flying...

Comment (masone) [reply backend/drone.js:38]: yeah, but it's `"command".length`, which will always be 7

File: backend/drone.js (hunk 1 above)
Comment (masone) [reply backend/drone.js:19]: oh, I see - drone state needs a different port than where we're sending the commands to. gotcha!

File: backend/drone.js (hunk 4 above)
Comment (MilosJovanovic1986) [reply backend/drone.js:38]: I mean `command.length` of each commands that drone have, like (`battery?` have 8, `land` have 4 and so on)
commands

Comment (masone) [reply backend/drone.js:38]: oooh, I understand. thanks for the explanation - one needs to indeed send "command" as the first command and then the actual command itself

**Pull request merged**